### PR TITLE
[MCOMPILER-526] Upgrade to parent 39

### PR DESCRIPTION
**Pull request by gnodet**
- [MCOMPILER-526] Upgrade to parent 39
- [MCOMPILER-526] Reformat

Requires https://github.com/apache/maven-compiler-plugin/pull/176

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MCOMPILER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MCOMPILER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MCOMPILER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by psiroky** [comment]
Yes, please! Thank you for this change. `<personalRant>` the old style was really hurting my eyes `</personalRant>`

File: src/it/MCOMPILER-474_recompile-dependent-when-package/dependent-module/src/main/java/Main.java [hunk 1]
```diff
@@ -16,11 +16,10 @@
  * specific language governing permissions and limitations
  * under the License.
  */
+package foo;
```
Comment (slawekjaranowski) [inline src/it/MCOMPILER-474_recompile-dependent-when-package/dependent-module/src/main/java/Main.java:19]: Should we move file to `.../java/foo/Main.java` to be consistent with package name?

Comment (gnodet) [reply src/it/MCOMPILER-474_recompile-dependent-when-package/dependent-module/src/main/java/Main.java:19]: We could, but there are tons of files to move manually for no benefit, and those are just tests, so I left it the way it is.  are you fancy doing it ?  I did not found  a trivial way to do that. 
Fwiw, the java compiler puts the compiled classes in the correct `foo/Main.class` while the groovy compile does not, so I had to move the groovy ones.

File: src/it/MCOMPILER-503-processorpath-duplicated-deps/annotation-user/src/main/java/mcompiler503/SimpleObject.java [hunk 2]
```diff
@@ -1,3 +1,21 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (slawekjaranowski) [inline src/it/MCOMPILER-503-processorpath-duplicated-deps/annotation-user/src/main/java/mcompiler503/SimpleObject.java:2]: double license header 😄

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**